# HG changeset patch
# Parent 20a161dabb40729976748bf5b13d14f27b76b130
Bug 590411 - Sync/Weave is not included if FF is built against libxul sdk

diff --git a/browser/build.mk b/browser/build.mk
--- a/browser/build.mk
+++ b/browser/build.mk
@@ -40,16 +40,21 @@ include $(topsrcdir)/toolkit/toolkit-tie
 endif
 
 TIERS += app
 
 ifdef MOZ_EXTENSIONS
 tier_app_dirs += extensions
 endif
 
+ifdef MOZ_SERVICES_SYNC
+tier_app_dirs += services/crypto
+tier_app_dirs += services/sync
+endif
+
 tier_app_dirs += $(MOZ_BRANDING_DIRECTORY)
 
 tier_app_dirs += toolkit/components/console/hudservice
 
 tier_app_dirs += browser
 
 installer:
 	@$(MAKE) -C browser/installer installer
diff --git a/toolkit/toolkit-tiers.mk b/toolkit/toolkit-tiers.mk
--- a/toolkit/toolkit-tiers.mk
+++ b/toolkit/toolkit-tiers.mk
@@ -281,20 +281,15 @@ endif
 ifdef MOZ_LEAKY
 tier_platform_dirs        += tools/leaky
 endif
 
 ifdef MOZ_MAPINFO
 tier_platform_dirs	+= tools/codesighs
 endif
 
-ifdef MOZ_SERVICES_SYNC
-tier_platform_dirs += services/crypto
-tier_platform_dirs += services/sync
-endif
-
 ifdef ENABLE_TESTS
 tier_platform_dirs += testing/mochitest
 tier_platform_dirs += testing/xpcshell 
 tier_platform_dirs += testing/mozmill
 tier_platform_dirs += testing/tools/screenshot
 endif
 
