diff --git a/browser/base/content/credits.xhtml b/browser/base/content/credits.xhtml
--- a/browser/base/content/credits.xhtml
+++ b/browser/base/content/credits.xhtml
@@ -583,16 +583,19 @@
             <li>Kohei Yoshino</li>
             <li>Shigeru Yoshitake</li>
             <li>Tanner M. Young</li>
             <li>Boris Zbarsky</li>
             <li>Marco Zehe</li>
             <li>Matthew Zeier</li>
           </ul>
 
+          <h3>openSUSE RPM packaging</h3>
+          <ul><li>Wolfgang Rosenauer</li></ul>
+
           <div dir="&locale.dir;">
             &credit.translation;
           </div>
         </div>
 
         <!-- organizational supporters -->
         <div class="creditsGroup">
           <h3 dir="&locale.dir;">&credit.thanks;</h3>
