# HG changeset patch
# Parent  c5d7373760d7abbd34f89cb60d4a45cba2c05379
Part 2 of
Bug 1136958 - gfx/layers/basic/BasicCompositor.cpp:20:56: fatal error: skia/SkCanvas.h: No such file or directory when skia is disabled

diff --git a/gfx/skia/trunk/include/core/SkPreConfig.h b/gfx/skia/trunk/include/core/SkPreConfig.h
--- a/gfx/skia/trunk/include/core/SkPreConfig.h
+++ b/gfx/skia/trunk/include/core/SkPreConfig.h
@@ -87,16 +87,24 @@
 
 #if !defined(SK_WARN_UNUSED_RESULT)
     #define SK_WARN_UNUSED_RESULT __attribute__((warn_unused_result))
 #endif
 
 //////////////////////////////////////////////////////////////////////
 
 #if !defined(SK_CPU_BENDIAN) && !defined(SK_CPU_LENDIAN)
+    #if defined(__BIG_ENDIAN__)
+         #define SK_CPU_BENDIAN
+    #elif defined(__LITTLE_ENDIAN__)
+        #define SK_CPU_LENDIAN
+    #endif
+#endif
+
+#if !defined(SK_CPU_BENDIAN) && !defined(SK_CPU_LENDIAN)
     #if defined(__sparc) || defined(__sparc__) || \
       defined(_POWER) || defined(__powerpc__) || \
       defined(__ppc__) || defined(__hppa) || \
       defined(__PPC__) || defined(__PPC64__) || \
       defined(_MIPSEB) || defined(__ARMEB__) || \
       defined(__s390__) || \
       (defined(__sh__) && defined(__BIG_ENDIAN__)) || \
       (defined(__ia64) && defined(__BIG_ENDIAN__))
diff --git a/js/src/jsdtoa.cpp b/js/src/jsdtoa.cpp
--- a/js/src/jsdtoa.cpp
+++ b/js/src/jsdtoa.cpp
@@ -47,16 +47,17 @@ using namespace js;
 static inline void* dtoa_malloc(size_t size) { return js_malloc(size); }
 static inline void dtoa_free(void* p) { return js_free(p); }
 
 #define NO_GLOBAL_STATE
 #define NO_ERRNO
 #define MALLOC dtoa_malloc
 #define FREE dtoa_free
 #include "dtoa.c"
+#undef CONST
 
 /* Mapping of JSDToStrMode -> js_dtoa mode */
 static const uint8_t dtoaModes[] = {
     0,   /* DTOSTR_STANDARD */
     0,   /* DTOSTR_STANDARD_EXPONENTIAL, */
     3,   /* DTOSTR_FIXED, */
     2,   /* DTOSTR_EXPONENTIAL, */
     2};  /* DTOSTR_PRECISION */
